xarray/core/dataset.py

```diff
@@ -1205,24 +1209,28 @@ def indexes(self) -> 'Mapping[Any, pd.Index]':
         return Indexes(self._indexes)
 
     @property
-    def coords(self):
+    def coords(self) -> DatasetCoordinates:
         """Dictionary of xarray.DataArray objects corresponding to coordinate
         variables
         """
         return DatasetCoordinates(self)
 
     @property
-    def data_vars(self):
+    def data_vars(self) -> DataVariables:
         """Dictionary of DataArray objects corresponding to data variables
         """
         return DataVariables(self)
 
-    def set_coords(self, names, inplace=None):
+    def set_coords(
+        self,
+        names: 'Union[Hashable, Iterable[Hashable]]',
```

Comment:
It would have been nicer to use Collection[Hashable], but Collection requires Python 3.6

Reply:
Note: Sequence is NOT ok, because sets are not sequences